2-api/server.js

```diff
@@ -0,0 +1,227 @@
+const { query, json } = require("express");
+const express = require("express");
+const app = express();
+const { Pool } = require("pg");
+app.use(express.json());
+
+const pool = new Pool({
+  user: "Maziar",
+  host: "localhost",
+  database: "cyf_ecommerce",
+  password: "M@ziar13",
+  port: 5432,
+});
+
+app.get("/", function (req, res) {
+  res.send("Hi there, please type /customers , /suppliers , /products. Have Fun!");
+});
+
+app.get("/customers", function (req, res) {
+  pool
+    .query("SELECT * FROM customers")
+    .then((result) => res.json(result))
+    .catch((e) => console.error(e));
+});
+
+app.get("/suppliers", function (req, res) {
+  pool
+    .query("SELECT * FROM suppliers")
+    .then((result) => res.json(result))
+    .catch((e) => console.error(e));
+});
+
+app.get("/products", function (req, res) {
+  const productName = req.query.product_name;
+  let query =
+    "SELECT products.product_name, product_availability.unit_price, suppliers.supplier_name FROM ((products INNER JOIN product_availability ON products.id = product_availability.prod_id) INNER JOIN suppliers ON suppliers.id = product_availability.supp_id)";
+
+  if (productName) {
+    query = `SELECT * FROM products WHERE product_name LIKE '%${productName}%'`;
+  }
```

Comment:
I think you should add the "like" clause to the original query rather than do a select *